File: backend/Models/Quiz.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;


namespace prid_2324_a12.Models;


public enum Statut{
    PAS_COMMENCE= 0,
    OUVERT=1,
    EN_COURS=2,
    CLOTURE =3,
    FINI= 4,
    PUBLIE= 5,
    PAS_PUBLIE=6,
}

//rajouter la property statut et check en backend le statut
public class Quiz{
    [Key]
    public int Id {get;set;}
    public string Name{get;set;}= null!;
    public string? Description {get;set;}
    public bool IsPublished {get;set;}
    public bool IsClosed {get;set;}
    public bool IsTest {get; set;}
    public DateTimeOffset? Start {get;set;}
    public DateTimeOffset? Finish {get;set;}
    [ForeignKey(nameof(DatabaseId))]
    public Database Database {get;set;}= null!;
    public int DatabaseId {get;set;}
    

    public Statut Statut {get;set;}

    [NotMapped]
    public int Note{get;set;}
    [NotMapped]
    public bool CanInteract{get;set;}
    [NotMapped]
    public bool HaveAttempt{get;set;}
    // [ForeignKey(nameof(CreatorId))]
    // public User Creator {get;set;}
    // public int CreatorId {get;set;}
    //devrait avoir une seule db ??
   // public ICollection<Database> Databases {get;set;} = new HashSet<Database>();
    [JsonIgnore]
    public ICollection<Question> Questions {get;set;} = new HashSet<Question>();

    [JsonIgnore]
    public ICollection<Attempt> Attempts {get;set;} = new HashSet<Attempt>();

   public string getStatutasString(){
        switch(Statut)
        {
            case Statut.PAS_COMMENCE:
                return "PAS_COMMENCE";
            case Statut.OUVERT:
                return "OUVERT";
            case Statut.EN_COURS:
                return "EN_COURS";
            case Statut.CLOTURE:
                return "CLOTURE";
            case Statut.FINI:
                return "FINI";
            default:
                return "PAS_COMMENCE";
        }
   }
   public string getNote(){
        if(this.Note > 0)
            return this.Note + "/10";
        else
            return "N/A";
   }


}

File: frontend/src/app/models/quiz.ts
import { Type } from "class-transformer";
import 'reflect-metadata';
import { Question } from "./question";
import { Database } from "./database";

export enum Statut{
    PAS_COMMENCE= 0,
    OUVERT=1,
    EN_COURS=2,
    CLOTURE =3,
    FINI= 4,
    PUBLIE = 5,
    PAS_PUBLIE=6,
}
export class Quiz{
    id?: number;
    name?: string;
    description?: string;
    isPublished?: boolean;
    isClosed?: boolean;
    isTest?: boolean;
    haveAttempt?: boolean;
    canInteract?: boolean;
    start?: Date;
    finish?: Date;
    database?: Database;
    statut?: Statut;
    action?: string;
    note?:number;

    questions?: Question[];
    
    get display(): string {
        
        return `${this.name} ${this.database} ${this.getstatut}  ${this.action} ${this.note}`;
    }


    
    public get getstatut(): string {
        switch (this.statut) {
            case Statut.PAS_COMMENCE:
                return "PAS_COMMENCE";
            case Statut.OUVERT:
                return "OUVERT";
            case Statut.EN_COURS:
                return "EN_COURS";
            case Statut.CLOTURE:
                return "CLOTURE";
            case Statut.FINI:
                return "FINI";
            case Statut.PUBLIE:
                return "PUBLIE";
            case Statut.PAS_PUBLIE:
                    return "PAS_PUBLIE";
            default:
                return "PAS_COMMENCE";
        }
    }

    public get getNote() : string{
        if(this.note){
            return this.note+"/10";
        }
        return "N/A";
    }
}

export class QuizSave{
    DatabaseId?: number;
    Name?: string;
    Description?: string;
    IsPublished?: boolean;
    IsTest?: boolean;
    Start?: Date;
    Finish?: Date;
    Questions?: Question[];
   
}

export class QuizEdit{
    DatabaseId?: number;
    Name?: string;
    Description?: string;
    IsPublished?: boolean;
    IsTest?: boolean;
    Start?: Date;
    Finish?: Date;
    Questions?: Question[];
}